Book-A-Desk.Api/EmailNotification.fs

```diff
@@ -0,0 +1,46 @@
+namespace Book_A_Desk.Api
+
+open System
+open System.Net.Mail
+open Book_A_Desk.Api.Models
+open Book_A_Desk.Domain.QueriesHandler
+open Book_A_Desk.Domain.Office.Domain
+
+type EmailNotification =
+    {
+        SendEmailNotification: Booking -> unit
+    }
+
+module EmailNotification =
+    let initialize getEmailServiceConfiguration getOffices =         
+        let sendEmailNotification (booking: Booking) =
+            let config = getEmailServiceConfiguration()        
+            let officeName =
+                let result = OfficeQueriesHandler.getAll getOffices
+                let officeReference = Guid.Parse(booking.Office.Id) |> OfficeId
+                match result with
+                | Ok offices ->
+                    offices
+                    |> List.find (fun (o:Office) -> officeReference.Equals(o.Id))
+                    |> (fun (o:Office) -> 
+                        let (CityName cityName) = o.City
+                        cityName)
+                | Error e ->
+                    "Unknown"
+                    
+            use smtpClient = new SmtpClient(config.SmtpClientUrl)
```

Comment:
My suggestion: have different functions for getting the office name and sending an email. Then you can do:
let sendEmailNotification booking =
    getOfficeName booking
    |> sendMail config

Reply:
Makes it easy to know exactly what you're doing :). FYI, ~30 lines is very long for a F# function, try to keep them <10-15 lines if possible.

Reply:
Same here: As this is a temporary solution anyway and as we send really just a few emails for me this is acceptable for now.
In the future when this is handled by an explicit event handler we will do differently anyway.

Reply:
To Do: Move getOfficeName as a part of the domain instead of getting it here.